src/github.com/matrix-org/dendrite/syncapi/consumers/roomserver.go

```diff
@@ -96,7 +96,7 @@ func (s *Server) onMessage(msg *sarama.ConsumerMessage) error {
 		}).Panicf("roomserver output log: write event failure")
 		return nil
 	}
-	s.rp.OnNewEvent(&ev, types.StreamPosition(syncStreamPos))
+	s.notifier.OnNewEvent(&ev, types.StreamPosition(syncStreamPos))
```

Comment:
This used to use a request pool but now it uses a notifier directly? This feels like a change that is potentially deeper than I would expect, if a request pool is actually some kind of pool of requests as the name would suggest. What happened here?